contracts/SpokePool.sol

```diff
@@ -535,22 +561,22 @@ abstract contract SpokePool is
      * If the message is not empty, the recipient contract must implement handleV3AcrossMessage() or the fill will revert.
      */
     function depositV3(
-        address depositor,
-        address recipient,
-        address inputToken,
-        address outputToken,
+        bytes32 depositor,
+        bytes32 recipient,
+        bytes32 inputToken,
+        bytes32 outputToken,
         uint256 inputAmount,
         uint256 outputAmount,
         uint256 destinationChainId,
-        address exclusiveRelayer,
+        bytes32 exclusiveRelayer,
         uint32 quoteTimestamp,
         uint32 fillDeadline,
         uint32 exclusivityPeriod,
         bytes calldata message
     ) public payable override nonReentrant unpausedDeposits {
         // Check that deposit route is enabled for the input token. There are no checks required for the output token
         // which is pulled from the relayer at fill time and passed through this contract atomically to the recipient.
-        if (!enabledDepositRoutes[inputToken][destinationChainId]) revert DisabledRoute();
+        if (!enabledDepositRoutes[inputToken.toAddress()][destinationChainId]) revert DisabledRoute();
```

Comment:
Should we changed the data structure to:

`mapping(bytes32 => mapping(uint256 => bool)) public enabledDepositRoutes;`

Reply:
I think yes, as that would avoid unnecessary conversion, and it would still use the same storage slot for the mapped values.